engines/wintermute/detection.cpp

```diff
@@ -210,6 +214,214 @@ class WintermuteMetaEngine : public AdvancedMetaEngine {
 		pm.getSaveStateDesc(slot, retVal);
 		return retVal;
 	}
+
+	Common::KeymapArray initKeymaps(const char *target) const {
+		using namespace Common;
+
+		Keymap *engineKeyMap = new Keymap(Keymap::kKeymapTypeGame, "wintermute", "WinterMute Engine");
+
+		Action *act;
+		
+		act = new Action("LCLK", _("Left Click"));
+		act->setLeftClickEvent();
+		act->addDefaultInputMapping("MOUSE_LEFT");
+		act->addDefaultInputMapping("JOY_A");
+		engineKeyMap->addAction(act);
+
+		act = new Action("RCLK", _("Right Click"));
+		act->setRightClickEvent();
+		act->addDefaultInputMapping("MOUSE_RIGHT");
+		act->addDefaultInputMapping("JOY_B");
+		engineKeyMap->addAction(act);
+
+		act = new Action("RETURN", _("Confirm"));
+		act->setKeyEvent(KeyState(KEYCODE_RETURN, ASCII_RETURN));
+		act->addDefaultInputMapping("RETURN");
+		//TODO: Joy control? Does
+		engineKeyMap->addAction(act);
+
+		act = new Action("ESCAPE", _("Escape"));
+		act->setKeyEvent(KeyState(KEYCODE_ESCAPE, ASCII_ESCAPE));
+		act->addDefaultInputMapping("ESCAPE");
+		act->addDefaultInputMapping("JOY_Y");
+		engineKeyMap->addAction(act);
+
+		act = new Action(kStandardActionOpenDebugger, _("Open debugger"));
+		act->setCustomEngineActionEvent(kWintermuteActionOpenDebugger);
+		act->addDefaultInputMapping("C+d");
+		engineKeyMap->addAction(act);
+		
+		Common::KeymapArray result = Keymap::arrayOf(engineKeyMap);
+
+		Common::String targetString = target;
```

Comment:
The target is user-controlled. It's possible to edit it when adding a game. Is fetching the gameid from the configuration manager not working for you? `Common::String gameId = ConfMan.get("gameid", target);`

Reply:
Thanks! Never noticed "gameid" property before.